lib/sisimai/bite/email/einsundeins.rb

```diff
@@ -0,0 +1,143 @@
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :from    => %r/\A["]Mail Delivery System["]/,
+        :subject => %r/\AMail delivery failed: returning message to sender\z/,
+      }
+      Re1 = {
+        :begin   => %r/\AThis message was created automatically by mail delivery software/,
+        :error   => %r/\AFor the following reason:/,
+        :rfc822  => %r/\A--- The header of the original message is following/,
+        :endof   => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ReFailure = {
+        mesgtoobig: %r/Mail[ ]size[ ]limit[ ]exceeded/x,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return '1&1: http://www.1and1.de'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      # X-UI-Out-Filterresults: unknown:0;
+      def headerlist;  return []; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from 1&1
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['from']    =~ Re0[:from]
+        return nil unless mhead['subject'] =~ Re0[:subject]
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:'message-rfc822']).zero?` instead of `readcursor & Indicators[:'message-rfc822'] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->